src/Common/MoistThermodynamics/states.jl

```diff
@@ -185,8 +205,10 @@ function LiquidIcePotTempSHumEquil_given_pressure(
     p::FT,
     q_tot::FT,
     maxiter::Int = 30,
-    tol::FT = FT(1e-1),
+    temperature_tol::FT = FT(1e-3),
 ) where {FT <: Real}
+    _cp_d = FT(cp_d(param_set))
+    tol = ResidualTolerance(temperature_tol*_cp_d)
```

Comment:
```suggestion
    tol = ResidualTolerance(potential_temperature_tol)
```
(we minimize potential temperature residuals, right? So no cp/cv factors here)